Test Failure with "make integration-fail-fast" command

### Describe the bug
A clear and concise description of what the bug is. What command did you run? What happened?
I ran "make integration-fail-fast" command, it fails with 
```
test/integration/004_simple_snapshot_test/test_simple_snapshot.py:862: in _snapshot_and_assert_invalidated
    self.assertGreaterEqual(result[-1].replace(tzinfo=pytz.UTC), self._invalidated_snapshot_datetime)
AssertionError: datetime.datetime(2021, 4, 26, 9, 7, 58, 180921, tzinfo=<UTC>) not greater than or equal to datetime.datetime(2021, 4, 26, 16, 7, 56, 828247, tzinfo=<UTC>)
```

### Steps To Reproduce
run the command in the PST timezone.

### Expected behavior
I don't understand why this fails.

### Screenshots and log output
If applicable, add screenshots or log output to help explain your problem.

### System information
**Which database are you using dbt with?**
- [x] postgres
- [ ] redshift
- [ ] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)


**The output of `dbt --version`:**
pulled from develop branch.

**The operating system you're using:**
MAC OS darwin 

**The output of `python --version`:**
Python 3.8.8

### Additional context
Add any other context about the problem here.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,7 @@
 Contributors:
 - [@TeddyCr](https://github.com/TeddyCr) ([#3275](https://github.com/fishtown-analytics/dbt/pull/3275))
 - [@panasenco](https://github.com/panasenco) ([#3315](https://github.com/fishtown-analytics/dbt/pull/3315))
+- [@peiwangdb](https://github.com/peiwangdb) ([#3344](https://github.com/fishtown-analytics/dbt/pull/3344))
 
 ## dbt 0.20.0b1 (May 03, 2021)
 

--- a/test/integration/004_simple_snapshot_test/test_simple_snapshot.py
+++ b/test/integration/004_simple_snapshot_test/test_simple_snapshot.py
@@ -859,7 +859,7 @@ class TestSnapshotHardDelete(DBTIntegrationTest):
         for result in snapshotted[10:]:
             # result is a tuple, the dbt_valid_to column is the latest
             self.assertIsInstance(result[-1], datetime)
-            self.assertGreaterEqual(result[-1].replace(tzinfo=pytz.UTC), self._invalidated_snapshot_datetime)
+            self.assertGreaterEqual(result[-1].astimezone(pytz.UTC), self._invalidated_snapshot_datetime)
 
     def _revive_records(self):
         database = self.default_database
@@ -899,7 +899,7 @@ class TestSnapshotHardDelete(DBTIntegrationTest):
         for result in invalidated_records:
             # result is a tuple, the dbt_valid_to column is the latest
             self.assertIsInstance(result[1], datetime)
-            self.assertGreaterEqual(result[1].replace(tzinfo=pytz.UTC), self._invalidated_snapshot_datetime)
+            self.assertGreaterEqual(result[1].astimezone(pytz.UTC), self._invalidated_snapshot_datetime)
 
         # records which weren't revived (id != 10, 11)
         revived_records = self.run_sql(
@@ -921,5 +921,5 @@ class TestSnapshotHardDelete(DBTIntegrationTest):
             self.assertIsInstance(result[1], datetime)
             # there are milliseconds (part of microseconds in datetime objects) in the
             # invalidated_snapshot_datetime and not in result datetime so set the microseconds to 0
-            self.assertGreaterEqual(result[1].replace(tzinfo=pytz.UTC), self._invalidated_snapshot_datetime.replace(microsecond=0))
+            self.assertGreaterEqual(result[1].astimezone(pytz.UTC), self._invalidated_snapshot_datetime.replace(microsecond=0))
             self.assertIsNone(result[2])

